src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -59,4 +97,24 @@ public void modifySpeed(GetMaxSpeedEvent event, EntityRef player) {
     public void modifyJumpSpeed(AffectJumpForceEvent event, EntityRef player) {
         event.multiply(jumpSpeedMultiplier);
     }
+
+    private void applyFrostbiteDamagePlayer(EntityRef player) {
+        Prefab frostbiteDamagePrefab = prefabManager.getPrefab("ClimateConditions:FrostbiteDamage");
+        player.send(new DoDamageEvent(healthDecreaseAmount, frostbiteDamagePrefab));
+    }
+
+    private void applyStunEffect(EntityRef player, int duration) {
+        StunAlterationEffect stunAlterationEffect = new StunAlterationEffect(context);
+        //Both the instigator and the target is the player
+        //the magnitude parameter is not used by StunAlterationEffect
+        stunAlterationEffect.applyEffect(player, player, 0, duration);
```

Comment:
That .... works? I still don't understand AlterationEffects - do you @jdrueckert?

Reply:
From what we have seen till yet, AlterationEffects work completely fine as long as they are applied for a definite duration. The only trouble is when we want them to go on for an indefinite duration.